src/classes/VOL_SharedCode.cls

```diff
@@ -400,66 +400,67 @@ global with sharing class VOL_SharedCode {
                 strSoql += strComma + strF;
                 strComma = ', ';
             }
-        }        
+        }
         strSoql += ' from Contact ';
         string strAnd = ' where ';
-        
+
         // make sure their settings haven't been completely cleared
         if (VolunteersSettings.Contact_Matching_Rule__c == null || VolunteersSettings.Contact_Matching_Rule__c == '') {
             VolunteersSettings.Contact_Matching_Rule__c = 'Firstname;Lastname;Email';
         }
-        
-        if (isLastnameInContactMatchRules) {
-            strSoql += strAnd + ' (Lastname=\'' + StrEscape(contact.Lastname) + '\') ';
+
+        final String rule = VolunteersSettings.Contact_Matching_Rule__c;
+        if (rule.containsIgnoreCase(String.valueOf(Contact.LastName))) {

```

Comment:
The original methods check for null or blank strings. Should we wrap the next few blocks in `if (!String.isBlank(rule)){ }` or return early if there is no setting?

Reply:
I think that was maybe left over from an earlier version because on line 409 we set it to the default and prevent it from ever being null or blank.

Reply:
Ah, I see. Would it make more sense to combine the lines?

`final String rule = (!String.isBlank(VolunteersSettings.Contact_Matching_Rule__c)) ? VolunteersSettings.Contact_Matching_Rule__c : 'FirstName;LastName';`

Or if you're not into ternaries, at least switch line 409 to String.isBlank?

Reply:
Then we would no longer be assigning the Default Contact Matching to the Volunteers Settings, which might be ok but I'm not sure where or when it is referenced after this method is called.

Reply:
That works!